x-pack/solutions/observability/plugins/observability/server/services/related_dashboards_client.ts

```diff
@@ -284,10 +308,43 @@ export class RelatedDashboardsClient {
     return fields;
   }
 
+  async getLinkedDashboards(): Promise<RelatedDashboard[]> {
+    if (!this.alert) {
+      throw new Error('Alert not found. Could not get linked dashboards.');
+    }
+    const ruleId = this.alert.getRuleId();
+    if (!ruleId) {
+      this.logger.warn(`Rule id not found. No linked dashboards available.`);
+      return [];
+    }
```

Comment:
I've been wondering about this; is there a reason we can simply log a warning and return an empty set here while we need to throw if the alert is `null`?